src/psyclone/dynamo0p3.py

```diff
@@ -5886,13 +5886,14 @@ def _create_depth_list(halo_info_list, sym_table):
                 match = True
                 break
         if not match:
-            # no matches were found with existing entries so
-            # create a new one
-            depth_info = HaloDepth(sym_table)
-            depth_info.set_by_value(max_depth=False, var_depth=var_depth,
-                                    literal_depth=literal_depth,
-                                    annexed_only=False, max_depth_m1=False)
-            depth_info_list.append(depth_info)
+            # no matches were found with existing entries so create a
+            # new one (unless no var depth and literal_depth is 0).
```

Comment:
I would suggest "unless there is no 'var_depth' and 'literal_depth' is 0)."

Reply:
Updated.